specs/altair/p2p-interface.md

```diff
@@ -140,6 +140,8 @@ def get_sync_subcommittee_pubkeys(state: BeaconState, subcommittee_index: uint64
 - _[REJECT]_ `contribution_and_proof.selection_proof` selects the validator as an aggregator for the slot -- i.e. `is_sync_committee_aggregator(contribution_and_proof.selection_proof)` returns `True`.
 - _[REJECT]_ The aggregator's validator index is in the declared subcommittee of the current sync committee --
   i.e. `state.validators[contribution_and_proof.aggregator_index].pubkey in get_sync_subcommittee_pubkeys(state, contribution.subcommittee_index)`.
+- _[IGNORE]_ The block being signed (`contribution_and_proof.contribution.beacon_block_root`) has been seen (via both gossip and non-gossip sources) (a client MAY queue sync committee contributions for processing once block is received)
```

Comment:
I'm not sure about adding this in as a condition.  It's not uncommon to see a fair bit of spread around the time that a block is received across different beacon nodes on the same server, let alone across multiple servers.  Generating and signing sync committee messages is a pretty quick process, so if a beacon node discards these messages because it has yet to see the relevant block it could well be failing to hold on to a correct message.

How about if the beacon node will only ignore the message only if it has seen a block for the current slot?  That gives protection against this common situation.  If there is a concern about flooding the network with bogus messages by slower nodes then limit it to a single unknown beacon block root per beacon node.

Reply:
It's the same condition as for attestations, aggregates, and EIP-7732 payloads / attestations.

- https://github.com/ethereum/consensus-specs/blob/001caab3b34732420fdbddb0fff4cb50e21a84ef/specs/phase0/p2p-interface.md?plain=1#L340-L342

In practice, sync messages and attestations are sent around ~4s into the slot, and sync contributions and aggregates are sent around ~8s into the slot.

If the condition is a problem here, it is also a problem for attestations.

Reply:
The point is not the specific time that sync messages show up, it's that there can be a fair spread of time over which some nodes have the current slot's block and others do not.

From my understanding, at least some of the beacon nodes queue attestations when they do not have the block for the current slot rather than discard them (not sure if they forward them or not, that's another question).  But having nodes ignore sync committee messages for the current slot when the node doesn't yet have the current slot's block doesn't seem to be an optimal solution to me.

Reply:
My point is only that the processing should be done consistently for attestations as well as sync messages. If this is a problem for sync messages, it is also a problem for attestations. 

For attestations, the current spec wording has been in place since phase0. Implementations for example ensure that attestations are delayed a bit to ensure block propagation; and they use unknown roots as a trigger to req/resp the unknown block. Sync messages follow the same code path, as in, are practically sent with a delay after processing the block, or at 4s into the slot. It _could_ be that the wording for attestations also does not properly reflect the reality; but the PR here solely ensures consistent handling of the two messages (attestations / sync messages) that sign over the head, it does not introduce new problems that may or may not be present with the existing logic.